libsel4vm/src/arch/arm/vgic/gicv3.h

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright 2017, Data61, CSIRO (ABN 41 687 119 230)
+ * Copyright 2019, DornerWorks
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#define GIC_500_GRP0     BIT(0)
+#define GIC_500_GRP1_NS  BIT(1)
+#define GIC_500_GRP1_S   BIT(2)
+#define GIC_500_ARE_S    BIT(4)
+
+#define GIC_500_ENABLED (GIC_500_ARE_S | GIC_500_GRP1_NS | GIC_500_GRP0)
+#define GIC_ENABLED GIC_500_ENABLED
+
+#define GIC_SGI_OFFSET  0x10000
+#define GIC_MAX_DISABLE 32
+#define GIC_GROUP       1
+#define GIC_V3
+
+
+#define GIC_DIST_PADDR    0x38800000
+#define GIC_REDIST_PADDR  0x38880000
+
+struct gic_dist_map {
+    uint32_t ctlr;                /* 0x0000 */
+    uint32_t typer;               /* 0x0004 */
+    uint32_t iidr;                /* 0x0008 */
+    uint32_t res1[13];            /* [0x000C, 0x0040) */
+    uint32_t setspi_nsr;          /* 0x0040 */
+    uint32_t res2;                /* 0x0044 */
+    uint32_t clrspi_nsr;          /* 0x0048 */
+    uint32_t res3;                /* 0x004C */
+    uint32_t setspi_sr;           /* 0x0050 */
+    uint32_t res4;                /* 0x0054 */
+    uint32_t clrspi_sr;           /* 0x0058 */
+    uint32_t res5[9];             /* [0x005C, 0x0080) */
+    uint32_t irq_group0[CONFIG_MAX_NUM_NODES];          /* [0x080, 0x84) */
```

Comment:
This copies the same sloppiness than in the GICv2. We are not adding 1 filed here, but multiple in case there are multiple codes. So the offsets mentioned in the comments are not really right. It does not matter practically, but it might be a bit misleading of one really does offset calculation here - they wont always work. But that's a separate thing to clean up one day

Reply:
This code was originally written several years ago. I've tried to fold through the updates to the original files as I've rebased it, but some things may have been missed.  Thanks for pointing out any changes that were missed.